File: language/scala/src/main/resources/scala-code/tree/Node.scala
package tree

class Node(val children:Seq[Tree], val label:Int) extends Tree {

  // copy elements
  val subtrees:Seq[Tree] = children.map {x => x}

  override def canEqual(a: Any) = a.isInstanceOf[Node]

  /** link in with default equals. */
  override def equals(tree: Any): Boolean = {
    tree match {
      case o: Node => o.canEqual(this) && same(o)

      case _ => false
    }
  }

  override def asNode():Option[Node] = Some(this)
}

File: src/main/resources/java-code/Node.java
public class Node implements Tree {
    public final int label;
    java.util.List<Tree> subtrees = new java.util.ArrayList<Tree>();

    public boolean equals (Object o) {
        if (o == null) { return false; }
        if (o instanceof Tree) {
            return same((Tree) o);
        }
        return false;
    }

    public Node(java.util.List<Tree> children, int label) {
        this.label = label;
        subtrees.addAll(children);
    }

    public java.util.Optional<Node> asNode() { return java.util.Optional.of(this); }

    public String toString() {
        String children = "";
        for (Tree t : subtrees) {
            children += t.toString() + ", ";
        }
        return "[Label=" + label + ", " + children + "]";
    }
}